app/workers/builds.php

```diff
@@ -248,22 +249,25 @@ protected function buildDeployment(GitHub $github, Document $project, Document $
                     );
                 }
 
-                Console::execute('tar --exclude code.tar.gz -czf /tmp/builds/' . \escapeshellcmd($buildId) . '/code.tar.gz -C /tmp/builds/' . \escapeshellcmd($buildId) . '/code' . (empty($rootDirectory) ? '' : '/' . $rootDirectory) . ' .', '', $stdout, $stderr);
+                $tmpPath = '/tmp/builds/' . \escapeshellcmd($buildId);
+                $tmpPathFile = $tmpPath . '/code.tar.gz';
 
-                $deviceFunctions = $this->getFunctionsDevice($project->getId());
+                Console::execute('tar --exclude code.tar.gz -czf ' . $tmpPathFile . ' -C /tmp/builds/' . \escapeshellcmd($buildId) . '/code' . (empty($rootDirectory) ? '' : '/' . $rootDirectory) . ' .', '', $stdout, $stderr);
 
-                $fileName = 'code.tar.gz';
-                $fileTmpName = '/tmp/builds/' . $buildId . '/code.tar.gz';
+                $deviceFunctions = $this->getFunctionsDevice($project->getId());
 
-                $path = $deviceFunctions->getPath($deployment->getId() . '.' . \pathinfo($fileName, PATHINFO_EXTENSION));
+                $localDevice = new Local();
+                $buffer = $localDevice->read($tmpPathFile);
```

Comment:
This could be dangerous if tar files are larger than the max memory allowed, right?

I think we're okay using `move()` because [`move()` calls `transfer()`](https://github.com/utopia-php/storage/blob/efec5376c02d3d8330f1beb1469e6d6e313e21ee/src/Storage/Device.php#L165) and `transfer()` will [chunk](https://github.com/utopia-php/storage/blob/efec5376c02d3d8330f1beb1469e6d6e313e21ee/src/Storage/Device/S3.php#L293-L299) if needed.

Reply:
Actually, move will still move from the provider so we can't use that.

Reply:
We'll need to handle chunking later